src/EnergyPlus/DataRootFinder.hh

```diff
@@ -54,124 +54,102 @@
-    // F(XMin) == F(XMax)
-    extern int const iStatusErrorSingular;
-    // Error because the overall slope assumption is not observed at the min and max points:
-    // - for an increasing function F(X), we expect F(XMin) < F(XMax)  otherwise error
-    // - for a decreasing function F(X),  we expect F(XMin) > F(XMax)  otherwise error
-    // Note that this error status does not detect strict monotonicity at points
-    // between the min and max points.
-    extern int const iStatusErrorSlope;
-    // Error because the current candidate X does not lie within the current lower an upper points:
-    // X < XLower or X > XUpper
-    extern int const iStatusErrorBracket;
-    // Error because the current candidate X does not lie within the min and max points:
-    // X < XMin or X > XMax
-    extern int const iStatusErrorRange;
-
-    extern int const iStatusNone; // Indeterminate error state (not converged), also default state
-    extern int const iStatusOK;   // Unconstrained convergence achieved with root solution so that:
-    // XMin < XRoot < XMax
-    extern int const iStatusOKMin;      // Constrained convergence achieved with solution XRoot==XMin
-    extern int const iStatusOKMax;      // Constrained convergence achieved with solution XRoot==XMax
-    extern int const iStatusOKRoundOff; // Reached requested tolerance in X variables although Y=F(X) does not
-    // satisfy unconstrained convergence check
-
-    extern int const iStatusWarningNonMonotonic; // Error because F(X) is not strictly monotonic between the
-    // lower and upper points
-    extern int const iStatusWarningSingular; // Error because F(X) == YLower or F(X) == YUpper
-
-    extern int const iMethodNone;          // No solution method (used internally only when root finder is reset)
-    extern int const iMethodBracket;       // Bracketting mode (used internally only to bracket root)
-    extern int const iMethodBisection;     // Step performed using bisection method (aka interval halving)
-    extern int const iMethodFalsePosition; // Step performed using false position method (aka regula falsi)
-    extern int const iMethodSecant;        // Step performed using secant method
-    extern int const iMethodBrent;         // Step performed using Brent's method
-    // Names for each solution method type
-    extern Array1D_string const SolutionMethodTypes;
-
-    // DERIVED TYPE DEFINITIONS
-    // Type declaration for the numerical controls.
-
-    // Type declaration for iterate tracking.
-
-    // Type declaration for the root finder solution technique.
-
-    // Types
-
-    struct ControlsType
+namespace EnergyPlus::DataRootFinder {
+
+constexpr int iSlopeNone(0);        // Undefined slope specification
+constexpr int iSlopeIncreasing(1);  // For overall increasing function F(X) between min and max points
+constexpr int iSlopeDecreasing(-1); // For overall decreasing function F(X) between min and max points
```

Comment:
@amirroth @Myoldmopar These were not changed to an enum class because the values are used in error reporting messages, which frankly are not super useful. But that's what we've got. For example: https://github.com/NREL/EnergyPlus/blob/83402156ae773d8b87568ffac3f867fe68b1910b/src/EnergyPlus/RootFinder.cc#L244-L245 
I don't expect either of you to propose a solution at this time, but just noting FFR. We'll run into this as we push more getting these converted over, so we might think start thinking about how situations like this might be handled.

Reply:
Thx.